### PR TITLE
Wizard: Expose step value to querystring

### PR DESCRIPTION
**Pull request by SquidwardHill**
Signed-off-by: Sydney Hill <sydneyehill@gmail.com>

### CODE REVIEW

File: packages/forms/src/Components/Wizard.php [hunk 1]
```diff
@@ -16,10 +16,14 @@ class Wizard extends Component
 
     protected bool | Closure $skippable = false;
 
+    protected bool | Closure $stepAsQuerystring = false;
+
     protected string | Htmlable | null $submitAction = null;
 
     public int | Closure $startStep = 1;
 
+    public ?int $currentStep = null;
```
Comment (danharrin) [inline packages/forms/src/Components/Wizard.php]: I think this is not being used?

File: packages/forms/src/Components/Wizard.php [hunk 2]
```diff
@@ -16,10 +16,14 @@ class Wizard extends Component
 
     protected bool | Closure $skippable = false;
 
+    protected bool | Closure $stepAsQuerystring = false;
```
Comment (danharrin) [inline packages/forms/src/Components/Wizard.php]:
```suggestion
    protected bool | Closure $isStepInQueryString = false;
```

File: packages/forms/src/Components/Wizard.php [hunk 3]
```diff
@@ -100,6 +104,13 @@ public function skippable(bool | Closure $condition = true): static
         return $this;
     }
 
+    public function setStepAsQuerystring(bool | Closure $condition = true): static
```
Comment (danharrin) [inline packages/forms/src/Components/Wizard.php]:
```suggestion
    public function queryStringStep(bool | Closure $condition = true): static
```

File: packages/forms/src/Components/Wizard.php [hunk 4]
```diff
@@ -119,4 +130,9 @@ public function isSkippable(): bool
     {
         return $this->evaluate($this->skippable);
     }
+
+    public function isStepAsQuerystring(): bool
```
Comment (danharrin) [inline packages/forms/src/Components/Wizard.php]:
```suggestion
    public function isStepInQueryString(): bool
```

**Review by danharrin — CHANGES_REQUESTED** [review]
You might need to rename other method and property references to match these

**Comment by danharrin** [comment]
@zepfietje @awcodes wdyt about naming that I proposed? it's difficult

**Comment by awcodes** [comment]
I think they are ok.

**Comment by zepfietje** [comment]
I'd prefer something like `storeStepInQueryString()` with `shouldStoreStepInQueryString()`. Quite lengthy though.

Also, I'm wondering what's the purpose of storing the step in the query string if the step is not restored from the query string when loading the page anyway?

**Comment by danharrin** [comment]
How about `persistStepInQueryString()`? And yes, it should be restored on load

**Comment by zepfietje** [comment]
I like that. Think it matches the naming around table filters.

**Comment by danharrin** [comment]
Okay, lets use that naming then @SquidwardHill 

`persistStepInQueryString()`
`$isStepPersistedInQueryString` / `isStepPersistedInQueryString()`

**Comment by zepfietje** [comment]
`isPersisted` or `shouldPersist`, @danharrin?

**Comment by danharrin** [comment]
I try and use `is` wherever it makes grammatical sense. `should` is a last resort

**Comment by zepfietje** [comment]
Makes sense! Still waiting for your method naming docs. 😉

**Comment by lhilton** [comment]
> Also, I'm wondering what's the purpose of storing the step in the query string if the step is not restored from the query string when loading the page anyway?

Speaking to this question, I can add some context: When we developed this PR, the decision was to make the minimal filament-side change possible and let the handling of that be on the consumer side. In other words, leave it to the end forms to determine the right way to respond to that querystring. I can see a case for going the other route, however this would require more substantial design in order to both support the change and allow developer overrides of that behavior.

**Comment by zepfietje** [comment]
Thanks for clarifying, @lhilton! I think it's important to add the full functionality to Filament if we're going to add it.

**Comment by lhilton** [comment]
@danharrin @zepfietje I'll coordinate with @SquidwardHill to move our restoration logic in as part of this PR then. Thank you for the feedback!

**Comment by SquidwardHill** [comment]
Circling back to this on Monday.

**Comment by zepfietje** [comment]
Any update on this PR @lhilton @SquidwardHill?

**Comment by lhilton** [comment]
We have updated this PR with the requested changes (name changes) and we moved the logic to automatically restore step state if the feature is in use.

Additionally we found a need to modify the querystring key in use. This was previously hardcoded to `step`. It will now default to the value `step` but can be overridden when using the `persistStepInQueryString()` function as an optional second parameter.

**Comment by danharrin** [comment]
Thanks!

**Pull request merged**